lib/publiccloud/provider.pm

```diff
@@ -384,22 +385,25 @@ sub terraform_apply {
     record_info('TFM cmd', $cmd);
 
     script_retry($cmd, timeout => $terraform_timeout, delay => 3, retry => 6);
-    my $ret = script_run('terraform apply -no-color -input=false myplan', $terraform_timeout);
-    unless (defined $ret) {
-        if (is_serial_terminal()) {
-            type_string(qq(\c\\));    # Send QUIT signal
+    my $tries = (is_azure() && get_var('PUBLIC_CLOUD_QAM')) ? 3 : 1;    # Retry only on azure platform for QAM runs
```

Comment:
What do you think if we add the retry count as a parameter to `terraform_apply()` function. 
I would like to avoid here things like `is_azure() and `get_var(PUBLIC_CLOUD_QAM)` treeting this more like a lib then test code. WDYT?
I know there is already `PUBLIC_CLOUD_SLES4SAP` but we could try to not make it more complicated.

Reply:
The retries could be set for azure default here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/azure.pm#L212

Reply:
Not sure about the exception for `QAM` why not retry always?